s64da_benchmark_toolkit/reporting.py

```diff
@@ -123,6 +123,8 @@ def _save_explain_plan(self, query_metric):
 
     def _save_prepare_metrics(self):
         if os.path.exists(self.prepare_metrics_path):
+            if not os.path.exists(self.results_root_dir):
```

Comment:
This nested `if` seems bit smelly to me. Why is it OK to not have `prepare_metrics_path`?

Reply:
prepare_metrics_path is always created during a benchmark run (function run in prepare.py), so the first if is obsolete

Reply:
Ok obviously the file does not exist in the tests. I will revert to a cascaded if for now

Reply:
Err, rather fix the test?

Reply:
Actually - it can happen that prepare_metrics_path is not there. For example when the toolkit benchmark is executed and the dataset was not loaded from the same toolkit checkout folder, or the file was moved/deleted in between.

If there is no objection I will bring back both IFs and revert the change in the test